### PR TITLE
[WIP] add bitmex support

### LINKED ISSUE #74: Add Bitmex Support
https://www.bitmex.com/

Rotkehlchen has a common exchange interface class seen [here](https://github.com/rotkehlchenio/rotkehlchen/blob/85cd33e2953273087bd2ee504f9c2bd9ed0adcba/rotkehlchen/exchange.py).

To add bitmex (or any other) exchange to rotkehlchen all that needs to be done is implement that interface for the new exchange.

A good example of how to do that can be seen in the [bittrex](https://github.com/rotkehlchenio/rotkehlchen/blob/master/rotkehlchen/bittrex.py) class.

Then in all the places where bittrex is used you should also add the new exchange.

### PR DESCRIPTION
**Pull request by PumpkingWok**
##### Description

- This PR implements the methods to connect to bitmex exchange using own api key and secret.
- I have followed the code logic of `Poloniex.py`, i have not tested some methods because not used.
- I can modify some parts of code if they are not well implemented.
- I have used `bitmex` official python api wrapper but it does not work so good, specially with zero parameter methods.

##### Checklist

- [X] Changes don't break existing behavior

##### Affected core subsystem(s)

Core

##### Screenshots (Optional) 

image 2018-08-28 18 06 35
image 2018-08-28 18 06 47

##### Testing

I have tested it in Mac OS environment inside conda env.

##### Refers/Fixes

Fixes #74

### CODE REVIEW

File: rotkehlchen/bitmex.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def tsToDate(s):
+    return datetime.datetime.fromtimestamp(s).strftime('%Y-%m-%d %H:%M:%S')
+
+
+class Bitmex(Exchange):
```
Comment (LefterisJP) [inline rotkehlchen/bitmex.py]: This entire file is wrong. I will try to explain in the comments of the PR.

**Comment by LefterisJP** [comment]
@PumpkingWok What were you trying to do with the bitmex file? It's a copy and paste of poloniex.py. The names of the function even contain the rest API calls to poloniex and not bitmex.

On top of that poloniex.py has **a lot** of unnecessary code and calls that are no longer used which is why I told you not to look there too long. For reference please look at what [bittrex](https://github.com/rotkehlchenio/rotkehlchen/blob/master/rotkehlchen/bittrex.py) does and where its class is used. It's same as poloniex but without all the extra stuff.

As I have already [described](https://github.com/rotkehlchenio/rotkehlchen/issues/74#issuecomment-416040353) in the issue there is a [common interface](https://github.com/rotkehlchenio/rotkehlchen/blob/master/rotkehlchen/exchange.py) class which all exchanges should implement. 

That is **all you have to do** for bitmex. Nothing more. Look at bittrex as an example and figure out how to do the same for bitmex. [Here](https://www.bitmex.com/app/restAPI) is the API documentation of bitmex. That is what you should use, not some wrapper.

**Comment by PumpkingWok** [comment]
I have followed the poloniex code flow as i told you, i have tried to include all methods, even not used.only the name of api is the same name than poloniex api but until the bitmex wrapper call the function with the exact name.have you tested the code ? The description of this issue is a link.do i need to follow bittrex.py structure ? Thanks

**Comment by LefterisJP** [comment]
Do you understand what an API is?

>  i have tried to include all methods, even not used.

Please do not do that.

> the name of api is the same name than poloniex api but until the bitmex wrapper call the function with the exact name

Why on earth would you do that?

> have you tested the code ?

No I don't need to. It's not what we need.

> do i need to follow bittrex.py structure ?

You need to implement the bitmex API that satisfies the exchange interface as I already explained both above and in the issue.

**Comment by PumpkingWok** [comment]
The description of this issue is composed of one link, i understand well what API is. You did not tell in the description that i could not use a official bitmex wrapper.i accomplished the issue, i will leave the issue now, you are a crazy man !! You need to learn how create a github issue, with description !!

**Pull request closed without merging**